### PR TITLE
Send all errors to STDERR

### PR DESCRIPTION
**Pull request by scjody**
This makes all errors go to STDOUT consistently, and future-proofs us if Dokku adds fancier error handling later.

@bpostlethwaite Please review.
@josegonzalez Feel free to take a look if you like.

### CODE REVIEW

File: pre-build [hunk 1]
```diff
@@ -27,7 +28,7 @@ fi
 
 if [[ ! -d "$ACL" ]]; then
   if [[ -n "$DOKKU_SUPER_USER" ]] && [[ "$NAME" != "$DOKKU_SUPER_USER" ]]; then
-    echo "Only $DOKKU_SUPER_USER can modify a repository if the ACL is empty" >&2;
+    dokku_log_fail "Only $DOKKU_SUPER_USER can modify a repository if the ACL is empty"
     exit 1;
```
Comment (josegonzalez) [inline pre-build]: Don't need this.

File: pre-build [hunk 2]
```diff
@@ -37,6 +38,6 @@ fi
 ACL_FILE="$ACL/$NAME"
 
 if [[ ! -f "$ACL_FILE" ]] && [[ "$NAME" != "$DOKKU_SUPER_USER" ]]; then
-  echo "User $NAME does not have permissions to modify this repository" >&2
+  dokku_log_fail "User $NAME does not have permissions to modify this repository"
   exit 2;
```
Comment (josegonzalez) [inline pre-build]: This won't exit 2 anymore.

File: user-auth [hunk 3]
```diff
@@ -27,23 +27,23 @@ done
...
 
-    echo "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist" >&2
+    dokku_log_fail "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist"
     exit 1
   fi
 done
 
-echo "User $SSH_NAME does not have permissions to run $CMD"
+dokku_log_fail "User $SSH_NAME does not have permissions to run $CMD"
 exit 1
```
Comment (josegonzalez) [inline user-auth]: Don't need this.

File: user-auth [hunk 4]
```diff
@@ -27,23 +27,23 @@ done
...
+      dokku_log_fail "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist"
       exit 1
     fi
 
     ACL_FILE="$DOKKU_ROOT/$APP/acl/$SSH_NAME"
     [[ -f "$ACL_FILE" ]] && exit 0
 
-    echo "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist" >&2
+    dokku_log_fail "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist"
     exit 1
```
Comment (josegonzalez) [inline user-auth]: Don't need this.

File: user-auth [hunk 5]
```diff
@@ -27,23 +27,23 @@ done
...
-      echo "An app name is required"
+      dokku_log_fail "An app name is required"
       exit 1
     fi
 
     APP=$2
     if ! ( verify_app_name "$APP" 2>/dev/null ); then
-      echo "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist" >&2
+      dokku_log_fail "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist"
       exit 1
```
Comment (josegonzalez) [inline user-auth]: Don't need this.

File: user-auth [hunk 6]
```diff
@@ -27,23 +27,23 @@ done
 for allowed in $DOKKU_ACL_PER_APP_COMMANDS; do
   if [[ "$CMD" == "$allowed" ]]; then
     if [[ -z "$2" ]]; then
-      echo "An app name is required"
+      dokku_log_fail "An app name is required"
       exit 1
```
Comment (josegonzalez) [inline user-auth]: Don't need this.

File: pre-build [hunk 7]
```diff
@@ -14,10 +15,10 @@ if [[ -z "$NAME" ]] ; then
...
-    echo -n "It appears that you're running this command from the command "
-    echo -n "line.  The \"dokku-acl\" plugin disables this by default for "
-    echo -n "safety.  Please check the \"dokku-acl\" documentation for how "
-    echo "to enable command line usage."
+    dokku_log_fail "It appears that you're running this command from the command" \
+      "line.  The \"dokku-acl\" plugin disables this by default for" \
+      "safety.  Please check the \"dokku-acl\" documentation for how" \
+      "to enable command line usage."
 
     exit 1
```
Comment (josegonzalez) [inline pre-build]: Don't need this.

File: commands [hunk 8]
```diff
@@ -66,7 +69,7 @@ case "${1:-}" in
 
     if [ -z "$USER" ];
     then
-      echo "Please specify a user name" >&2
+      dokku_log_fail "Please specify a user name"
       exit 1;
```
Comment (josegonzalez) [inline commands]: Don't need this.

File: commands [hunk 9]
```diff
@@ -55,7 +58,7 @@ case "${1:-}" in
     ACL_FILE="$ACL_PATH/$USER"
     if [ -f "$ACL_FILE" ];
     then
-       echo "User already has permissions to push to this repository"
+       dokku_log_fail "User already has permissions to push to this repository"
        exit 2;
```
Comment (josegonzalez) [inline commands]: This won't be called.

File: commands [hunk 10]
```diff
@@ -46,7 +49,7 @@ case "${1:-}" in
 
     if [ -z "$USER" ];
     then
-      echo "Please specify a user name" >&2
+      dokku_log_fail "Please specify a user name"
       exit 1;
```
Comment (josegonzalez) [inline commands]: Don't need this.

File: commands [hunk 11]
```diff
@@ -12,21 +15,21 @@ ACL_PATH="$DOKKU_ROOT/$APP/acl"
...
+    dokku_log_fail "No such app: $1"
     echo 2;
     exit;
   fi
 
   if [ -n "${NAME:-}" ];
   then
-    echo "You can only modify ACL using local dokku command on target host" >&2
+    dokku_log_fail "You can only modify ACL using local dokku command on target host"
     echo 3;
```
Comment (josegonzalez) [inline commands]: This and the exit won't be called.

File: commands [hunk 12]
```diff
@@ -12,21 +15,21 @@ ACL_PATH="$DOKKU_ROOT/$APP/acl"
...
+    dokku_log_fail "You must provide an app name!"
     echo 1;
     exit;
   fi
 
   if [ ! -d "$DOKKU_ROOT/$1" ];
   then
-    echo "No such app: $1" >&2
+    dokku_log_fail "No such app: $1"
     echo 2;
```
Comment (josegonzalez) [inline commands]: This and the exit won't be called.

File: commands [hunk 13]
```diff
@@ -12,21 +15,21 @@ ACL_PATH="$DOKKU_ROOT/$APP/acl"
 check_app () {
   if [ -z "$1" ];
   then
-    echo "You must provide an app name!" >&2
+    dokku_log_fail "You must provide an app name!"
     echo 1;
```
Comment (josegonzalez) [inline commands]: This and the exit won't be called.

**Comment by scjody** [comment]
@josegonzalez Sorry, I missed the `exit 1` in `dokku_log_fail`. I fixed the issues and also refactored `check_app` since it was a bit strange. How does this look now?

@bpostlethwaite OK with you?

**Review by josegonzalez — APPROVED** [review]

**Comment by bpostlethwaite** [comment]
LGTM :dancer:

**Pull request merged**